fabric_plugin/tasks.py

```diff
@@ -562,6 +562,16 @@ def _make_proxy(ctx, port):
         proxy.close()
 
 
+def handle_sudo(conn, use_sudo, command):
+    # This is the only solution that works: https://stackoverflow.com/questions/54638426/python-fabric-sudo-su-user/54682223.  # noqa
+    if use_sudo and not PY2:
+        command = 'echo "{command}" | sudo su'.format(command=command)
+        run = conn.run
```

Comment:
I'm not sure I understand why we can't use conn.sudo- with a local test using the fabric under the systests framework it seems to work (for systests, fabric == 2.6.0):
`>>> from fabric import Connection
>>> conn = Connection(
...     host='127.0.0.1',
...     user='user',
...     port=22,
...     connect_timeout=3,
... )
>>> conn.open()
>>> dir(conn)
>>> conn.run('echo test')
test
<Result cmd='echo test' exited=0>
>>> conn.sudo('echo test')
test
<Result cmd="sudo -S -p '[sudo] password: ' echo test" exited=0>
>>>`

Reply:
(this'd be easier without markdown- that was from an interactive python interpreter)